loadtest/control/gencontroller/actions.go

```diff
@@ -416,13 +410,28 @@ func (c *GenController) joinChannel(u user.User) control.UserActionResponse {
 	if cm.UserId == "" {
 		// We use sysadmin to add channel in case it's a private channel.
 		// Otherwise normal users don't have permissions to join a private channel.
-		err = c.sysadmin.AddChannelMember(channelID, u.Store().Id())
+		err = c.user.AddChannelMember(channelID, u.Store().Id(), c.sysadmin.Client())
 		if err != nil {
+			// Since we are selecting channels from global state, sometimes
+			// we can pick a channel whose team the user has not joined. In that case,
+			// we join the team first.
+			if appErr := err.(*model.AppError); appErr != nil && appErr.Id == "app.team.get_member.missing.app_error" {
+				teamID, err := extractTeamIdFromError(appErr.DetailedError)
+				if err != nil {
+					return control.UserActionResponse{Err: control.NewUserError(err)}
+				}
+
+				err = c.joinTeamWithId(teamID, u)
+				if err != nil {
+					return control.UserActionResponse{Err: control.NewUserError(err)}
+				}
+				return control.UserActionResponse{Info: fmt.Sprintf("user wasn't a team member. Added to team %s", teamID)}
+			}
```

Comment:
What if instead of this, we just join all teams for all users? As an init action, replace the current `joinTeam` by a new `joinAllTeams`.

Reply:
Hmm .. but that'll only work with dbs with existing teams right? Or do you mean create all the necessary teams first, then join to all those teams, then create necessary channels, and then go on with the rest of the actions.

Reply:
Ah, that's true, I had not thought about that. But yes, we can
1. Create all teams
2. Join all teams
3. Create all channels
4. Do the rest

Reply:
Hmm .. this is getting more and more complicated.. Let's see what @streamer45 thinks.